sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -0,0 +1,307 @@
+
+template<typename T, typename PropertyListT>
+struct is_property_key_of<
+  cache_control_write_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+
+template<int L>
+inline constexpr cache_control_read_key::value_t<cache_control_read_type::cached, L>
+cache_control_read_cached;
+
+template<int L>
+inline constexpr cache_control_read_key::value_t<cache_control_read_type::uncached, L> cache_control_read_uncached;
+
+template<int L>
+inline constexpr cache_control_read_key::value_t<cache_control_read_type::streaming, L> cache_control_read_streaming;
+
+template<int L>
+inline constexpr cache_control_read_key::value_t<cache_control_read_type::invalidate_after_read, L> cache_control_invalidate_after_read;
+
+template<int L>
+inline constexpr cache_control_read_key::value_t<cache_control_read_type::const_cached, L> cache_control_read_const_cached;
+
+template<int L>
+inline constexpr cache_control_write_key::value_t<cache_control_write_type::uncached, L> cache_control_write_uncached;
+
+template<int L>
+inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_streaming, L> cache_control_write_streaming;
+
+template<int L>
+inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_through, L> cache_control_write_through;
+
+template<int L>
+inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_back, L> cache_control_write_back;
+
+} // namespace sycl::ext::intel::experimental
+```
+Each of these properties takes a cache level parameter indicating which level
+of the cache hierarchy is affected. Cache level 0 indicates the cache closest
+to the processing unit, cache level 1 indicates the next furthest cache
+level, etc. It is legal to specify a cache level that does not exist on
+the target device, but the property will be ignored in this case.
+
+Note that a property specifies the cache behavior only for the indicated
+cache level. In order to specify the behavior for multiple cache levels, 
+multiple properties should be specified.
+
+It is legal to specify several different cache control properties in the
+same `annotated_ptr`. However, all instances of cache_control_read_type must
+have different cache levels and all instances of cache_control_write_type
+must have difference cache levels.
+
+The cache control properties are divided into two categories: those that
+are hints and those that are assertions by the application.
+
+==== Cache control hints
+These properties are hints requesting specific cache behavior when
+loading or storing to memory through the annotated_ptr. These properties can
+affect the performance of device code, but they do not change the semantics.
+
+--
+[options="header", cols="2,1"]
+|====
+| Property | Description
+a|
+[source]
+----
+cache_control_read<cache_control_read_type::cached, L>
+----
+|
+This property requests that loads from memory through the `annotated_ptr`
+may cache the data at level L in the memory hierarchy.
```

Comment:
```suggestion
may cache the data at level `L` in the memory hierarchy.
```

The `L` should be code font since it's the name of a template parameter.  Same for the other entries below.

Reply:
OK.
Have also added a line in the Implementation notes that the cache controls could use compiler intrinsics.